src/Microsoft.AspNetCore.Antiforgery/AntiforgeryTokenSet.cs

```diff
@@ -15,28 +15,57 @@ public class AntiforgeryTokenSet
         /// </summary>
         /// <param name="requestToken">The token that is supplied in the request.</param>
         /// <param name="cookieToken">The token that is supplied in the request cookie.</param>
-        public AntiforgeryTokenSet(string requestToken, string cookieToken)
+        /// <param name="formFieldName">The name of the form field used for the request token.</param>
+        /// <param name="headerName">The name of the header used for the request token.</param>
+        public AntiforgeryTokenSet(
+            string requestToken,
+            string cookieToken,
+            string formFieldName,
+            string headerName)
         {
             if (string.IsNullOrEmpty(requestToken))
             {
                 throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(requestToken));
             }
 
+            if (string.IsNullOrEmpty(formFieldName))
+            {
+                throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(formFieldName));
+            }
+
+            if (string.IsNullOrEmpty(headerName))
```

Comment:
User can set `AntiforgeryOptions.HeaderName` to `null` or `string.Empty` w/o error. Further `null` is special cased later to mean "don't look for a header". So parameter validation is incorrect here.